File: javascript/src/models/listResponseStreamSinkOut.ts
// this file is @generated
import { type StreamSinkOut, StreamSinkOutSerializer } from "./streamSinkOut";

export interface ListResponseStreamSinkOut {
  data: StreamSinkOut[];
  iterator: string | null;
  prevIterator?: string | null;
  done: boolean;
}

export const ListResponseStreamSinkOutSerializer = {
  _fromJsonObject(object: any): ListResponseStreamSinkOut {
    return {
      data: object["data"].map((item: StreamSinkOut) =>
        StreamSinkOutSerializer._fromJsonObject(item)
      ),
      iterator: object["iterator"],
      prevIterator: object["prevIterator"],
      done: object["done"],
    };
  },

  _toJsonObject(self: ListResponseStreamSinkOut): any {
    return {
      data: self.data.map((item) => StreamSinkOutSerializer._toJsonObject(item)),
      iterator: self.iterator,
      prevIterator: self.prevIterator,
      done: self.done,
    };
  },
};

File: rust/src/models/list_response_stream_sink_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::stream_sink_out::StreamSinkOut;

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct ListResponseStreamSinkOut {
    pub data: Vec<StreamSinkOut>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub iterator: Option<String>,

    #[serde(rename = "prevIterator")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub prev_iterator: Option<String>,

    pub done: bool,
}

impl ListResponseStreamSinkOut {
    pub fn new(data: Vec<StreamSinkOut>, done: bool) -> Self {
        Self {
            data,
            iterator: None,
            prev_iterator: None,
            done,
        }
    }
}
